app/forms/candidate_interface/science_gcse_grade_form.rb

```diff
@@ -0,0 +1,236 @@
+    validates :biology_grade, presence: true, on: :grades, if: :triple_award?
+    validates :chemistry_grade, presence: true, on: :grades, if: :triple_award?
+    validates :physics_grade, presence: true, on: :grades, if: :triple_award?
+    validates :other_grade, presence: true, if: :grade_is_other?
+    validates :award_year, presence: true, on: :award_year
+    validates :grade, length: { maximum: 6 }, unless: :international_gcses_flag_active?, on: :grade
+    validate :grade_length, on: :grade
+    validate :award_year_is_a_valid_date, if: :award_year, on: :award_year
+    validate :validate_grade_format, unless: :new_record?, on: :grade
+    validate :triple_award_grades_format, if: :triple_award?, on: :grade
+
+    class << self
+      Grade = Struct.new(:value, :option)
+
+      def single_award_grades
+        SINGLE_GCSE_GRADES
+      end
+
+      def double_award_grades
+        DOUBLE_GCSE_GRADES
+      end
+
+      def build_from_qualification(qualification)
+        if FeatureFlag.active?('international_gcses') && qualification.qualification_type == 'non_uk'
+          new(
+            grade: qualification.set_grade,
+            other_grade: qualification.set_other_grade,
+            award_year: qualification.award_year,
+            qualification: qualification,
+          )
+        else
+          new(build_params_from(qualification))
+        end
+      end
+
+    private
+
+      def build_params_from(qualification)
+        params = {
+          gcse_science: qualification.subject,
+          subject: qualification.subject,
+          award_year: qualification.award_year,
+          qualification: qualification,
+        }
+
+        case qualification.subject
+        when ApplicationQualification::SCIENCE_SINGLE_AWARD
+          params[:single_award_grade] = qualification.grade
+        when ApplicationQualification::SCIENCE_DOUBLE_AWARD
+          params[:double_award_grade] = qualification.grade
+        when ApplicationQualification::SCIENCE_TRIPLE_AWARD
+          grades = qualification.grades
+          return unless grades
+
+          params[:biology_grade] = grades['biology']
+          params[:chemistry_grade] = grades['chemistry']
+          params[:physics_grade] = grades['physics']
+        when 'science'
+          params[:grade] = qualification.grade
+        end
+
+        params
+      end
+    end
+
+    def save_grades
+      if !valid?(:grade)
+        log_validation_errors(:grade)
+        return false
+      end
+
+      triple_award_grades = {
+        biology: biology_grade,
+        physics: physics_grade,
+        chemistry: chemistry_grade,
+      }
+
+      qualification.update(grade: set_grade, grades: triple_award_grades, subject: subject)
+    end
+
+    def save_year
+      if valid?(:award_year)
+        qualification.update(grade: set_grade, award_year: award_year)
+        return true
+      end
+
+      false
+    end
+
+  private
+
+    def award_year_is_not_in_the_future
```

Comment:
It might be too early for this, but thoughts on pulling these validations into a separate module/lib? They will/should be reused throughout all qualification validations. It would also makes this file a bit easier to parse.

Reply:
Yep good spot - I'll remove